Ibistic.Public.NSwag.Extensions/Ibistic.Public.NSwag.Extensions/ApiVersionQueryStringProcessor.cs

```diff
@@ -0,0 +1,68 @@
+﻿using Namotion.Reflection;
+using NSwag.Generation.Processors.Contexts;
+using System.Collections;
+using System.Linq;
+using System.Reflection;
+using NSwagProcessors = NSwag.Generation.Processors;
+
+namespace Ibistic.Public.NSwag.Extensions
+{
+    public class ApiVersionQueryStringProcessor : NSwagProcessors.IOperationProcessor
+    {
+        /// <summary>Processes an action controller trying to add its api version to the action query string path.</summary>
+        /// <param name="context">The processor context.</param>
+        /// <returns>true if the action should be added to the OpenApi specification.</returns>
+        public bool Process(OperationProcessorContext context)
+        {
+            var nswagApiVersionProcessor = context.Settings.OperationProcessors.TryGet<NSwagProcessors.ApiVersionProcessor>();
+            string[] filteredVersions = nswagApiVersionProcessor.IncludedVersions;
+
+            string[] actionApiVersions = GetActionApiVersions(context);
+
+            // If the action/controller has no the ApiVersion attribute
+            if (actionApiVersions.Length == 0)
+            {
+                // Include action in the OpenApi specification if the document is not being created with filtered versions.
+                return filteredVersions == null || filteredVersions.Length == 0;
+            }
+
+            // If no filter by versions, get the first (higher) one
+            if (filteredVersions == null || filteredVersions.Length == 0)
+            {
+                IncludeApiVersionInQueryString(context, actionApiVersions[0]);
+                return true;
+            }
+
+            foreach (string actionApiVersion in actionApiVersions)
+            {
+                // The first matching is considered the preferred api version usage
+                if (nswagApiVersionProcessor.IncludedVersions.Contains(actionApiVersion))
```

Comment:
Use the variable filteredVersions instead of repeating nswagApiVersionProcessor.IncludedVersions

Reply:
Done.